shared/entity/type.go

```diff
@@ -2,19 +2,24 @@ package entity
 
 import (
 	"fmt"
-	"net/url"
-	"runtime"
-	"strings"
-
-	"github.com/canonical/lxd/shared"
-	"github.com/canonical/lxd/shared/api"
-	"github.com/canonical/lxd/shared/logger"
-	"github.com/canonical/lxd/shared/version"
 )
 
 // Type represents a resource type in LXD that is addressable via the API.
 type Type string
 
+// typeInfo represents common attributes an entity type must have.
+//
+// To create a new entity type, add a new const Type, then create a type that implements typeInfo and add it to the
+// entityTypes map.
+type typeInfo interface {
+	// requiresProject returns whether the Type requires a project to be uniquely specified, e.g. true if it is project
+	// specific, false if not.
+	requiresProject() bool
+
+	// path returns the API path for the resource. The pathPlaceholder should be used in place of mux variables.
```

Comment:
what is pathPlaceholder? 

path is a slice of strings, could you update this comment to explain what each slice element represents?

Reply:
ah pathPlaceholder is a constant, could you say "The pathPlaceholder constant should be used in place of mux variables."

Reply:
Done.